paddle/fluid/pybind/tensor_py.h

```diff
@@ -531,5 +428,90 @@ inline void PyCUDAPinnedTensorSetFromArray(
 }
 #endif
 
+namespace details {
+
+template <typename T>
+constexpr bool IsValidDTypeToPyArray() {
+  return false;
+}
+
+#define DECLARE_VALID_DTYPE_TO_PY_ARRAY(type)    \
+  template <>                                    \
+  constexpr bool IsValidDTypeToPyArray<type>() { \
+    return true;                                 \
+  }
+
+DECLARE_VALID_DTYPE_TO_PY_ARRAY(float);
```

Comment:
1. If we are supporting Numpy for this change. Had better to support basic Numpy types.
https://www.numpy.org/devdocs/user/basics.types.html
See the section of non-platform-dependent list: np.int8, np.int16, np.int32 ...

2. Suggest to sort these so you can see some types are missed. For example, uint16_t, uint32_t etc.

Reply:
1. I just follow the previous logic. Not all data types were supported previously. See [here](https://github.com/PaddlePaddle/Paddle/pull/17182/files#diff-e63f2a3aed0a7dc4f649dabb8272344dL134).

2. Done.

Reply:
Sure, LGTM, also